src/app/webui/components/hooks/useChat.ts

```
@@ -0,0 +1,250 @@
+    // Content for text-based messages
+    content: string | null | Array<TextPart | ImagePart>;
+    // Optional image data sent directly in user messages
+    imageData?: { base64: string; mimeType: string };
+    // Tool call and result metadata
+    toolName?: string;
+    toolArgs?: any;
+    toolResult?: any;
+}
+
+export function useChat(wsUrl: string) {
+    const wsRef = useRef<WebSocket | null>(null);
```

ESLint

[error] 'WebSocket' is not defined.

(no-undef)